lib/mongoid/history/trackable.rb

```diff
@@ -3,8 +3,26 @@ module History
     module Trackable
       extend ActiveSupport::Concern
 
+      def self.included(mod)
+        # re-parse options at end of class definition
+        TracePoint.trace(:end) do |tp|
+          if mod == tp.self && mod.track_history_options
+            options_parser = Mongoid::History::Options.new(mod)
```

Comment:
This feels duplicate from code below, I think we want to move this into something like `@parsed ||= parse_track_history_options` or similar.